src/pal/src/thread/process.cpp

```diff
@@ -4313,43 +4310,47 @@ getPath(
     if (lpwstr)
     {
         /* convert path to multibyte, check buffer size */
-        n = WideCharToMultiByte(CP_ACP, 0, lpwstr, -1, lpPathFileName, iLen,
+        n = WideCharToMultiByte(CP_ACP, 0, lpwstr, -1, NULL, 0,
             NULL, NULL);
-        if (n == 0)
+        
+        if (!lpPathFileName.Reserve(n + lpFileNameString.GetCount() ))
         {
-            ASSERT("WideCharToMultiByte failure!\n");
+            ERROR("StackString Reserve failed!\n");
             return FALSE;
         }
 
-        n += strlen(lpFileName) + 2;
-        if (n > (INT)iLen)
-        {
-            ERROR("Buffer too small for full path!\n");
-            return FALSE;
-        }
+        lpPath = lpPathFileName.OpenStringBuffer(n);
+
+        n = WideCharToMultiByte(CP_ACP, 0, lpwstr, -1, lpPath, n,
+            NULL, NULL);
+        lpPathFileName.CloseBuffer(n - 1);
 
-        if ((strcat_s(lpPathFileName, iLen, "/") != SAFECRT_SUCCESS) ||
-            (strcat_s(lpPathFileName, iLen, lpFileName) != SAFECRT_SUCCESS))
+        if (n == 0)
         {
-            ERROR("strcat_s failed!\n");
+            ASSERT("WideCharToMultiByte failure!\n");
             return FALSE;
         }
 
+        lpPathFileName.Append("/", 1);
+        lpPathFileName.Append(lpFileNameString);
+
         if (access(lpPathFileName, F_OK) == 0)
         {
-            TRACE("found %s in application directory (%s)\n", lpFileName, lpPathFileName);
+            TRACE("found %s in application directory (%s)\n", lpFileName, lpPathFileName.GetString());
             return TRUE;
         }
     }
 
     /* then try the current directory */
-    if ((strcpy_s(lpPathFileName, iLen, "./") != SAFECRT_SUCCESS) ||
-        (strcat_s(lpPathFileName, iLen, lpFileName) != SAFECRT_SUCCESS))
+    if (!lpPathFileName.Reserve(lpFileNameString.GetCount() ))
```

Comment:
I am still adding a 100 characters on every resize on the heap, so these smaller appends should be fine